File: stacks/nodejs/nestjs/src/controllers/crypt/crypt.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { LoopStatus } from '../../models/loop-status';
import { loopCall } from '../../utils';
import { CryptService } from '../../services/crypt/crypt.service';

@Controller()
export class CryptController {
  constructor(private readonly cryptService: CryptService) {}

  @Get('/crypt')
  async getStatus(
    @Query('delay') delayInput,
    @Query('percentage') percentageInput,
  ): Promise<LoopStatus> {
    return loopCall(delayInput, percentageInput, this.cryptService.process);
  }
}

File: stacks/elixir/plug-cowboy/lib/plug_cowboy_app/controllers/crypt_controller.ex
defmodule Controllers.CryptController do
  import Services.Crypt
  import Utils.DelayedTask
  alias Models.LoopStatus
  @moduledoc false

  def hash(percentage, delay) do
    times = start(&process/0, percentage, delay)
    LoopStatus.ok(times > 0, times)
  end

end
